### PR TITLE
Trade with 0x

### PR DESCRIPTION
**Pull request by guanbinrui**
+ [x] Able to switch trade provider
+ [x] Refactor trader UI to support TradeComputed
+ [x] Able to view trade with ZRX API
+ [x] Able to send ZRX swapping tx

settings
settings_2

### CODE REVIEW

File: packages/maskbook/src/plugins/Trader/types/trader.ts [hunk 1]
```diff
@@ -29,6 +32,25 @@ export enum TradePool {
...
+    estimatedOutputAmount: BigNumber
+    computed: {
+        maximumSold: BigNumber
+        minimumReceived: BigNumber
+        priceImpact: BigNumber
+        executionPrice: BigNumber
+        fee: BigNumber
+    }
+    path?: (EtherTokenDetailed | ERC20TokenDetailed)[]
+    trade_?: T
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Trader/types/trader.ts:51]: Is it a typo or an internal property/

Comment (guanbinrui) [reply packages/maskbook/src/plugins/Trader/types/trader.ts:53]: Not a typo. This field refs to the original trade object for accessing unlisted stuff in the computed trade.

File: packages/maskbook/src/utils/shadow-root/ShadowRootMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import '../../utils/shadow-root/ShadowRootPortal'
+import { PortalShadowRoot } from '../../utils/shadow-root/ShadowRootPortal'
+import { useSheetsRegistryStyles } from './renderInShadowRoot'
+import { ErrorBoundary } from '../../components/shared/ErrorBoundary'
+
+/**
+ * This is the low-level API to rendering a menu.
+ * Please use InjectedMenu if possible.
+ */
+export default function ShadowRootMenu(_props: MenuProps) {
```
Comment (Jack-Works) [inline packages/maskbook/src/utils/shadow-root/ShadowRootMenu.tsx:12]: IIRC we have a `useMenu` handle ShadowRoot correctly.

Comment (guanbinrui) [reply packages/maskbook/src/utils/shadow-root/ShadowRootMenu.tsx:12]: Not working on the content page.

File: packages/maskbook/src/extension/options-page/DashboardDialogs/Feedback.tsx [hunk 3]
```diff
@@ -43,7 +43,7 @@ export function DashboardFeedbackDialog(props: WrappedDialogProps) {
                             label={t('your_message')}
                             value={message}
                             onChange={(e) => setMessage(e.target.value)}
-                            variant="standard"
+                            variant="outlined"
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/options-page/DashboardDialogs/Feedback.tsx:46]: 🤔 i found many changes like this. is this a rebase mistake?

Comment (guanbinrui) [reply packages/maskbook/src/extension/options-page/DashboardDialogs/Feedback.tsx:46]: I found this change does not follow the original style of text boxes.

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**